src/reactviews/pages/QueryResult/queryResultStateProvider.tsx

```diff
@@ -41,7 +42,45 @@ const QueryResultStateProvider: React.FC<QueryResultContextProps> = ({
                             tabId: tabId,
                         });
                     },
+                    getExecutionPlan: function (
+                        planFile: ep.ExecutionPlanGraphInfo,
+                    ): Promise<ep.GetExecutionPlanResult> {
+                        webViewState?.extensionRpc.action("getExecutionPlan", {
+                            sqlPlanContent: planFile.graphFileContent,
+                        });
+
+                        if (webViewState?.state.executionPlan) {
+                            return Promise.reject(
+                                new Error("Execution plan is undefined"),
+                            );
+                        }
+
+                        return Promise.resolve(
+                            webViewState!.state!.executionPlan!,
```

Comment:
So many assertions... These are generally code smells for me - if you have to assert because the type checker didn't catch something, it's _usually_ a good indication that the types aren't quite set up right, or that you should add explicit checks for null/undefined  to help it along.  Not always, but usually.